### DISCUSSION TITLE
Question: What if I execute `docker run` in workflow?

### DISCUSSION THREAD

**Discussion by kingluo** [nektos/act#1361] [General]
It seems that the act itself runs in docker, not like the real github workflow uses vm.
What if I execute `docker run` in workflow yaml (It's a common case, because many CI depends on docker or docker-compose to setup test env)? Does it involve "docker in docker" issue?

If so, could act run container from containerd directly just like what [kind](https://kind.sigs.k8s.io/docs/design/initial) does?

**Comment by KnisterPeter** [comment]
Act mounts the docker socket of the host and starts new containers on the host thereof.

**Reply by kingluo** [reply]
https://github.com/nektos/act/blob/e1b906813e9053c57ef1c9f7d535532cd7924602/pkg/runner/run_context.go#L86

Here, right?

Side question, if all containers comes from host, then how act manage the network to allow access between the container where the workflow resides in and other dependent containers correctly?
